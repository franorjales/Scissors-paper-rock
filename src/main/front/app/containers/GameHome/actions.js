/*
 *
 * GameHome actions
 *
 */

import {
  PLAY_MATCH,
  RESTART_GAME,
  PUT_GAME,
  PUT_ERROR,
  PUT_HISTORICAL,
  GET_HISTORICAL,
  CHANGE_HISTORICAL_SCREEN_ON,
} from './constants';

export function playMatchAction(user) {
  return {
    type: PLAY_MATCH,
    user,
  };
}

export function restartGameAction(user) {
  return {
    type: RESTART_GAME,
    user,
  };
}

export function putGame(game) {
  return {
    type: PUT_GAME,
    game,
  };
}

export function putError(error) {
  return {
    type: PUT_ERROR,
    error,
  };
}

export function getHistoricalAction() {
  return {
    type: GET_HISTORICAL,
  };
}

export function putHistorical(historicalGames) {
  return {
    type: PUT_HISTORICAL,
    historicalGames,
  };
}

export function changeHistoricalScreenOnAction() {
  return {
    type: CHANGE_HISTORICAL_SCREEN_ON,
  };
}
