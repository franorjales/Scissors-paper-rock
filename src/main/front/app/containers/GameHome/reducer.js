/*
 *
 * GameHome reducer
 *
 */
import produce from 'immer';
import {
  PUT_GAME,
  PUT_ERROR,
  PUT_HISTORICAL,
  CHANGE_HISTORICAL_SCREEN_ON,
} from './constants';

export const initialState = {
  game: {
    user: null,
    matches: [],
    numberOfmatchesPlayeds: 0,
  },
  error: {
    errorStatus: null,
  },
  historicalGames: {
    totalRoundsPlayed: 0,
    totalWinsForPlayerOne: 0,
    totalWinsForPlayerTwo: 0,
    totalDraws: 0,
  },
  historicalGamesScreenOn: false,
};

/* eslint-disable default-case, no-param-reassign */
const gameHomeReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case PUT_GAME:
        draft.game = action.game;
        break;
      case PUT_ERROR:
        draft.error = action.error;
        break;
      case PUT_HISTORICAL:
        draft.historicalGames = action.historicalGames;
        draft.historicalGamesScreenOn = true;
        break;
      case CHANGE_HISTORICAL_SCREEN_ON:
        draft.historicalGamesScreenOn = false;
        break;
    }
  });

export default gameHomeReducer;
