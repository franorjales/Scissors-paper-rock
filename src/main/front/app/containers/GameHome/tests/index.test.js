/**
 *
 * Tests for GameHome
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import configureStore from '../../../configureStore';
import { playMatchAction, restartGameAction } from '../actions';
import { initialState } from '../reducer';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { GameHome } from '../index';

describe('<GameHome />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <GameHome
            gameHome={initialState}
            playMatch={playMatchAction}
            restartGame={restartGameAction}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
