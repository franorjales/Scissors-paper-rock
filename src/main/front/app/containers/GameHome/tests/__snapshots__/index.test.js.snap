// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameHome /> should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root makeStyles-root-1 MuiGrid-container"
>
  <div
    align="center"
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <h1>
      Hello! Do you want to play a game?
    </h1>
  </div>
  <div
    align="center"
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Play one Match!
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    align="center"
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Access to game history
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    align="center"
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
  />
</div>
`;
